### PR TITLE
Java 10

### PR DESCRIPTION
**Pull request by hadim**
DO NOT MERGE: need #34 to be merged first.

### CODE REVIEW

**Comment by sodre** [comment]
@conda-forge-admin, please rerender

**Comment by sodre** [comment]
@conda-forge-admin, please rerender.

**Comment by sodre** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -30,20 +30,16 @@ build:
   track_features:
     - openjdk
 
-requirements:
-  build:
-    - curl  # [unix]
-
 test:
+  requires:
+    - {{ compiler('c') }}
```
Comment (jakirkham) [inline recipe/meta.yaml]: IDK if selectors work here, but we might want to restrict this to Unix as we don't have a Windows JNI test yet.

Comment (sodre) [reply recipe/meta.yaml]: It looks like the toolchain was added, so the selector worked for the test section.

**Comment by sodre** [comment]
@hadim, would you mind incorporating fixes for #37 and #38 before merging this back into the master branch?

Also, I noticed that other folders are now available in the Java Distribution, for example the `conf` folder, can you make sure if that is needed and if so, where it should be placed. 

Thanks so much for refreshing this recipe!

File: recipe/meta.yaml (hunk 1 above)
Comment (hadim) [reply recipe/meta.yaml]: Should I restrict this to linux and osx?

**Comment by hadim** [comment]
Ready for review/merge @sodre

**Pull request merged**